test/unittest/render-cmdlets.ts

```diff
@@ -0,0 +1,122 @@
+                },
+                {
+                    id: '/AttestationProviders/patch/AttestationProviders_Update',
+                    name: 'attestation_provider_update',
+                    lines: [
+                        'az attestation attestation-provider update ',
+                        '--provider-name "myattestationprovider" ',
+                        '--resource-group "MyResourceGroup" ',
+                    ],
+                    lastLine: '--tags Property1="Value1" Property2="Value2" Property3="Value3"',
+                },
+                {
+                    id: '/AttestationProviders/delete/AttestationProviders_Delete',
+                    name: 'attestation_provider_delete',
+                    lines: [
+                        'az attestation attestation-provider delete -y ',
+                        '--provider-name "myattestationprovider" ',
+                    ],
+                    lastLine: '--resource-group "sample-resource-group"',
+                },
+            ],
+        },
+    };
+
+    it('renderPositive', async () => {
+        const tmplPath = path.join(
+            `${__dirname}`,
+            '../../src/templates/tests/cmdlet/test_positive.py.njx',
+        );
+        nunjucks.configure({ autoescape: false });
+        data.testData.className = 'PositiveTest';
+        let result = nunjucks.render(tmplPath, data);
+        const oriFile = path.join(
+            `${__dirname}`,
+            '../../test/unittest/expected/tests/cmdlet/ori_test_positive.py',
+        );
+        await writeFile(oriFile, result);
+        result = await readFile(oriFile);
+        const expectedFile = path.join(
+            `${__dirname}`,
+            '../../test/unittest/expected/tests/cmdlet/test_positive.py',
+        );
+        const expected = await readFile(expectedFile);
+        assert.deepStrictEqual(result, expected, 'render logic in test_positive.py is incorrect');
+        await rmFile(oriFile);
+    });
+
+    it('renderNegative', async () => {
+        const tmplPath = path.join(
+            `${__dirname}`,
+            '../../src/templates/tests/cmdlet/test_negative.py.njx',
+        );
+        nunjucks.configure({ autoescape: false });
+        data.testData.className = 'NegativeTest';
+        let result = nunjucks.render(tmplPath, data);
+        const oriFile = path.join(
+            `${__dirname}`,
+            '../../test/unittest/expected/tests/cmdlet/ori_test_negative.py',
+        );
+        await writeFile(oriFile, result);
+        result = await readFile(oriFile);
+        const expectedFile = path.join(
+            `${__dirname}`,
+            '../../test/unittest/expected/tests/cmdlet/test_negative.py',
+        );
+        const expected = await readFile(expectedFile);
+        assert.deepStrictEqual(result, expected, 'render logic in test_negative.py is incorrect');
+        await rmFile(oriFile);
+    });
+});
```

Comment:
Could we add some corner case such as, when we get empty data as no swagger example was provided or multiple operations point to the same example or multi examples refered by the same operation.

Reply:
Added an empty case, seem multi operations point to same example and multi examples referred by same operation handled by modelerfour/codemodel so don't have relation with this unit